### PR TITLE
Theme Fields: load is_block_theme only for < WP 6.3

### PR DESCRIPTION
**Pull request by creativecoder**
## Proposed changes:

* Adds version check for loading is_block_theme field to wp/v2/themes API response, since this field is added by WP 6.3 and above

See related Core ticket: https://core.trac.wordpress.org/ticket/58123#ticket

**Note:** I'm AFK for quite a while starting next week, so feel free to take over this PR and merge it. Thanks!

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

See previous PR: https://github.com/Automattic/jetpack/pull/30144/files

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Use REST API Tester from Jetpack debugger to check the response of `wp/v2/themes`. It should include `is_block_theme` from Jetpack if WP version is < 6.3, and from WP when the version is 6.3 or above.

### CODE REVIEW

File: projects/plugins/jetpack/_inc/lib/core-api/wpcom-fields/theme-fields-is-block-theme.php [hunk 1]
```diff
@@ -96,6 +96,9 @@ public function get_schema() {
 	}
 }
 
-// TODO: If this field is also added to Core, add a conditional to load only when using an older version of WordPress.
+// `is_block_theme` was added to the v2/themes endpoint in WordPress 6.3.
 // See https://core.trac.wordpress.org/ticket/58123
-wpcom_rest_api_v2_load_plugin( 'WPCOM_REST_API_V2_Theme_Fields_Is_Block_Theme' );
+global $wp_version;
+if ( version_compare( $wp_version, '6.3', '<' ) ) {
```
Comment (jeherve) [inline projects/plugins/jetpack/_inc/lib/core-api/wpcom-fields/theme-fields-is-block-theme.php]: Do we need to care about 6.3 Beta as well?

Comment (kraftbj) [reply projects/plugins/jetpack/_inc/lib/core-api/wpcom-fields/theme-fields-is-block-theme.php]: With 6.3 out now, let's not worry about the beta.

Comment (kraftbj) [inline projects/plugins/jetpack/_inc/lib/core-api/wpcom-fields/theme-fields-is-block-theme.php]:
```suggestion
if ( version_compare( $wp_version, '6.3', '<' ) ) { // @todo Remove when WordPress 6.3 is the minimum.
```

**Comment by sdixon194** [comment]
Took a look at this one, but couldn't tell what kind of response to look for. I just saw the `is_block_theme: true` response, but wasn't sure how to tell where the response was coming from 🤔

**Review by anomiex — APPROVED** [review]
"is_block_theme" is present with both 6.2 and 6.3. Seems good enough to me.

**Comment by sdixon194** [comment]
Cherry picked to release branch in c81e958de5a02483c3c5e507873c1eb5e1fb2b1b

**Comment by creativecoder** [comment]
Thanks for completing this one while I was out!

**Pull request merged**